### PR TITLE
Kpng ebpf backend POC

### PR DESCRIPTION
**Pull request by astoycos**
This PR contains the code implementing an POC of an ebpf based backend for KPNG.  Currently it supports only clusterIP TCP and UDP services, however additional functionality will be added in the future. See the backend's README.md for more information. 

Additionally this makes some changes/ fixes to other parts of the KPNG codebase including... 

- Implement `setup()` for the fullstate client
- Using j2 to have smarter yaml templating since we need to spin up additional debugging containers for the ebpf backend 
- Fixes a bug in the lightdiffstore tests 
- Adds support for building and running the ebpf backend locally with KIND

### CODE REVIEW

**Comment by jayunit100** [comment]
wait, why are no tests running?

**Comment by astoycos** [comment]
Tests are running now

**Comment by astoycos** [comment]
Updated to 
- Stop tracking bytecode 
- Compile BPF on the fly in the `hack/test-e2e.sh` script

**Comment by tgraf** [comment]
This is awesome! Super exciting.

> The design for the ebpf program was taken from
> Cilium's proxy design found here:
> https://github.com/cilium/cilium/blob/master/bpf/bpf_sock.c#L479

Thanks for calling this out. It would be great to also keep the original copyright notice in the respective source files as per license requirements of the original code.

**Comment by astoycos** [comment]
After running the test suite on my local these are the only ones failing 

```
[91m[1mSummarizing 13 Failures:[0m

[91m[1m[Fail] [0m[90m[sig-network] Conntrack [0m[91m[1m[It] should be able to preserve UDP traffic when server pod cycles for a NodePort service [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/ginkgo/internal/leafnodes/runner.go:113[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be possible to connect to a service via ExternalIP when the external IP is not assigned to a node [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:1225[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be able to create a functioning NodePort service [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:1190[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should have session affinity timeout work for NodePort service [LinuxOnly] [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2877[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be rejected when no endpoints exist [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2032[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be able to switch session affinity for service with type clusterIP [LinuxOnly] [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:209[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be able to switch session affinity for NodePort service [LinuxOnly] [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2960[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be able to change the type from ExternalName to NodePort [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:1375[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should serve multiport endpoints from pods  [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:913[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should be able to update service type to NodePort listening on same port number but different protocols [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:1269[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should have session affinity work for service with type clusterIP [LinuxOnly] [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:209[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should have session affinity timeout work for service with type clusterIP [LinuxOnly] [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:209[0m

[91m[1m[Fail] [0m[90m[sig-network] Services [0m[91m[1m[It] should have session affinity work for NodePort service [LinuxOnly] [Conformance] [0m
[37m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2960[0m

[1m[91mRan 289 of 7042 Specs in 493.242 seconds[0m
[1m[91mFAIL![0m -- [32m[1m276 Passed[0m | [91m[1m13 Failed[0m | [33m[1m0 Pending[0m | [36m[1m6753 Skipped[0m

```

**Comment by astoycos** [comment]
Added that copyright @tgraf Thanks for taking a look!

File: client/localsink/fullstate/fullstate.go [hunk 1]
```diff
@@ -14,11 +14,13 @@ type ServiceEndpoints struct {
 }
 
 type Callback func(item <-chan *ServiceEndpoints)
+type Setup func()
 
 // EndpointsClient is a simple client to kube-proxy's Endpoints API.
 type Sink struct {
 	Config   *localsink.Config
 	Callback Callback
+	SetupFcn Setup
```
Comment (mcluseau) [inline client/localsink/fullstate/fullstate.go:23]: I think renaming this to `SetupFunc` would be more go-style

Comment (astoycos) [reply client/localsink/fullstate/fullstate.go]: Done

**Review by mcluseau — APPROVED** [review]

**Comment by astoycos** [comment]
This should be pretty much ready to go @jayunit100

**Pull request merged**